network/slashing/consumer.go

```diff
@@ -62,37 +61,77 @@ func (c *Consumer) logOffense(networkOffense string, violation *Violation) {
 	e.Msg(fmt.Sprintf("potential slashable offense: %s", violation.Err))
 
 	// capture unauthorized message count metric
-	c.metrics.OnUnauthorizedMessage(role, violation.MsgType, violation.Channel.String(), networkOffense)
+	c.metrics.OnUnauthorizedMessage(role, violation.MsgType, violation.Channel.String(), misbehavior.String())
 }
 
-// OnUnAuthorizedSenderError logs an error for unauthorized sender error.
-func (c *Consumer) OnUnAuthorizedSenderError(violation *Violation) {
-	c.logOffense(unAuthorizedSenderViolation, violation)
+// reportMisbehavior reports the slashing violation to the alsp misbehavior report manager. When violation identity
+// is nil this indicates the misbehavior occurred either on a public network and the identity of the sender is unknown
+// we can skip reporting the misbehavior.
+// Args:
+// - misbehavior: the network misbehavior.
+// - violation: the slashing violation.
+// Any error encountered while creating the misbehavior report is considered irrecoverable and will result in a fatal log.
+func (c *Consumer) reportMisbehavior(misbehavior network.Misbehavior, violation *network.Violation) {
+	if violation.Identity == nil {
+		c.log.Debug().
+			Bool(logging.KeySuspicious, true).
+			Str("peerID", violation.PeerID).
+			Msg("violation identity unknown (or public) skipping misbehavior reporting")
+		c.metrics.OnViolationReportSkipped()
+		return
+	}
+	report, err := alsp.NewMisbehaviorReport(violation.Identity.NodeID, misbehavior)
+	if err != nil {
+		c.log.Fatal().
+			Err(err).
+			Bool(logging.KeySuspicious, true).
```

Comment:
this isn't really suspicious since any error is from a misconfigured argument passed in.

looking at the code, this isn't actually possible here as the code is written, but it could be in the future. Can you add a comment about how this failure would occur?

Reply:
https://github.com/onflow/flow-go/pull/4512/commits/b68dcecdaca86c2c2c9be273b9623832a2f53668